src/services/polling.service.ts

```diff
@@ -0,0 +1,201 @@
+import { TransactionResponse, TransactionStatus } from "fireblocks-sdk";
+import { Clients } from "../interfaces/Clients";
+import { patchTransactionAmountUsd } from "../util/cmc/patchTransactionAmountUsd";
+import {
+  handleTransactionCreated,
+  handleTransactionCreatedOrUpdated,
+  handleTransactionStatusUpdated,
+} from "./webhook.service";
+import { Transaction } from "../model/transaction";
+import { MoreThan } from "typeorm";
+import {
+  ITransactionDetails,
+  TransactionSubStatus,
+} from "../interfaces/transaction";
+import { fetchAllTxs } from "../util/fetch-all";
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+const groupBy = <T, K extends keyof never>(arr: T[], key: (i: T) => K) =>
+  arr.reduce(
+    (groups, item) => {
+      (groups[key(item)] ||= []).push(item);
+      return groups;
+    },
+    {} as Record<K, T[]>,
+  );
+
+export class PollingService {
```

Comment:
i personally prefer public methods above private ones, but up to you

Reply:
You got it